sql: stop populating legacy descriptor fields for FKs

RaduBerinde opened the pull request:
The `LegacyOriginIndex` and `LegacyReferencedIndex` fields of
`ForeignKeyConstraint` proto are written for backward compatibility with 19.2.
We no longer need them and there will be no value to set for the origin index
once we won't require one.

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/50591)
<!-- Reviewable:end -->

thoszhang reviewed (approved):
> Reviewed 4 of 4 files at r1.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/50591)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @jordanlewis)



<!-- Sent from Reviewable.io -->

RaduBerinde committed:
sql: stop populating legacy descriptor fields for FKs

The `LegacyOriginIndex` and `LegacyReferencedIndex` fields of
`ForeignKeyConstraint` proto are written for backward compatibility with 19.2.
We no longer need them and there will be no value to set for the origin index
once we won't require one.

Release note: None

RaduBerinde commented:
bors r+
